src/vm/method.hpp

```diff
@@ -108,7 +108,6 @@ enum MethodClassification
     mcInstantiated = 5, // Instantiated generic methods, including descriptors
                         // for both shared and unshared code (see InstantiatedMethodDesc)
 
-#ifdef FEATURE_COMINTEROP 
```

Comment:
This ifdef should stay.

Reply:
I guess somehow I need to know whether the method is com method or not.

Reply:
Where is it that you need that?

Reply:
The problem is that in line https://github.com/dotnet/coreclr/blob/430331c011a0460670e2743ab08caa36f914f52d/src/vm/methodtablebuilder.cpp#L3079, I need to choose a method type: the ideal case is to choose correct one instead of FCall.

Reply:
If I chose METHOD_TYPE_COMINTEROP, then I need to enable 
https://github.com/dotnet/coreclr/blob/32f0f9721afb584b4a14d69135bea7ddc129f755/src/vm/methodtablebuilder.inl#L516

Reply:
What's wrong with choosing FCall?

Reply:
that's why I remove #ifdef for mcComInterop

Reply:
I think you should be able to ifdef the whole COM special case:

```
#ifdef FEATURE_COMINTEROP
                if (fIsClassComImport
                    || GetHalfBakedClass()->IsProjectedFromWinRT()
                    || bmtProp->fComEventItfType
                    )
                {
...

                }
                else
#endif // FEATURE_COMINTEROP
                if (dwMethodRVA == 0)
                {
                    type = METHOD_TYPE_FCALL;
....
```

Reply:
Then these methods will be treated as METHOD_TYPE_FCALL. I am afraid some reflection API will give wrong answers when checking methods in COM Types.

Reply:
`MethodImplAttributes` are read directly from metadata. The method type does not influence them.

Reply:
And if we find a place that does not deal with it correctly, we will fix it up.

Reply:
Thanks a lot. just treat these methods as METHOD_TYPE_FCALL as suggested by you.